packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -7,61 +7,110 @@
+    }
+    return (
+        <Box>
+            <FormattedMessage
+                id="product_tile.price_display"
+                defaultMessage="
+                   {isMaster, select,
+                        true
+                            {
+                                {
+                                    isRange, select,
+                                        true {
+                                            {
+                                                isOnSale, select,
+                                                    true {<b>From</b>}
+                                                    fales {
+                                                        {
+                                                            hasRepresentedProduct, select,
+                                                                true {<b>From</b>}
+                                                                false {<span>From</span>}
+                                                                other {}
+                                                        }
+                                                    }
+                                                    other {<span>From</span>}
+                                            }
+                                        }
+                                        false {}
+                                        other {}
+                                }
+                            }
+                        false {}
+                        other {}
+                      }
+                   {isASet, select,
+                        true {From <span>{salePrice}</span>}
+                        false
+                            {
+                                {
+                                    isOnSale, select,
+                                         true {
+                                            <b><price>{salePrice}</price></b> <s><price>{listPrice}</price></s>
+                                         }
+                                         false {
+                                            {
+                                                hasRepresentedProduct, select,
+                                                    true  {<span><price>{salePrice}</price></span>}
+                                                    false {<b><price>{salePrice}</price></b>}
+                                                    other {<b><price>{salePrice}</price></b>}
+                                            }
+                                         }
+                                         other {<b><price>{salePrice}</price></b>}
+                                }
+
+                            }
+                        other {}
+                   }
+                "
+                values={{
+                    ...priceData,
+                    ...PRICE_DATA_CUSTOM_TAGS
+                }}
+            />
```

Comment:
Although we added whitespace here, unfortunately the translators would see what gets generated in the json file:

```json
"product_tile.price_display": {
	"defaultMessage": "{isMaster, select, true { { isRange, select, true {From} false {} other {} } } false {} other {} } {isASet, select, true {} false { { isOnSale, select, true { <b><price>{salePrice}</price></b> <s><price>{listPrice}</price></s> } false { { hasRepresentedProduct, select, true {<span><price>{salePrice}</price></span>} false {<b><price>{salePrice}</price></b>} other {<b><price>{salePrice}</price></b>} } } other {<b><price>{salePrice}</price></b>} } } other {} }"
},
```

What would they think of that long string? 😅 I think it'll be hard to read and error prone to update for them.

What we want them to translate are actually these possible results:
- `<b><price>{salePrice}</price></b> <s><price>{listPrice}</price></s>`
- `<span><price>{salePrice}</price></span>`
- `<b><price>{salePrice}</price></b>`
- and prepend “From”, if it’s a price range

We can separate our logic from the things that we want to be translated. For example, something like this:

```js
    const intl = useIntl()
    const messages = intl.defineMessages({
        // With "From"
        a: {
            defaultMessage:
                'From <b><price>{salePrice}</price></b> <s><price>{listPrice}</price></s>'
        },
        b: {defaultMessage: 'From <span><price>{salePrice}</price></span>'},
        c: {defaultMessage: 'From <b><price>{salePrice}</price></b>'},

        // Without "From"
        d: {
            defaultMessage: '<b><price>{salePrice}</price></b> <s><price>{listPrice}</price></s>',
            description: 'No need to translate'
        },
        e: {
            defaultMessage: '<span><price>{salePrice}</price></span>',
            description: 'No need to translate'
        },
        f: {
            defaultMessage: '<b><price>{salePrice}</price></b>',
            description: 'No need to translate'
        }
    })

    // Our logic goes here
    if (priceData.isMaster && priceData.isRange) {
        return intl.formatMessage(messages.a)
    } else if (...) {
        // ...
    }
```